acquisitions/src/main/resources/thunderjet/mod-orders/features/close-order-when-fully-paid-and-received.feature

```diff
@@ -1,84 +1,98 @@
 
     * configure headers = headersUser
+    * configure retry = { count: 4, interval: 1000 }
 
-    # load global variables
     * callonce variables
 
-    * def createOrder = read('classpath:thunderjet/mod-orders/reusable/create-order.feature')
-    * def createOrderLine = read('classpath:thunderjet/mod-orders/reusable/create-order-line.feature')
-    * def openOrder = read('classpath:thunderjet/mod-orders/reusable/open-order.feature')
-
-    * def orderId = callonce uuid1
-    * def poLineId = callonce uuid2
 
-    * configure retry = { count: 4, interval: 1000 }
-
-
-  Scenario: Create composite order
+  Scenario: Close order when fully paid and received
+    # 1. Create order, order line and open order
+    * def orderId = call uuid
+    * def poLineId = call uuid
     * def v = call createOrder { id: #(orderId) }
+    * def v = call createOrderLine { id: #(poLineId), orderId: #(orderId), fundId: #(globalFundId) }
+    * def v = call openOrder { orderId: "#(orderId)" }
 
-  Scenario: Create order line
-    * def v = callonce createOrderLine { id: #(poLineId), orderId: #(orderId), fundId: #(globalFundId) }
-
-  Scenario: Open order
-    * def v = callonce openOrder { orderId: "#(orderId)" }
-
-  Scenario: Get poLine and update payment and receipt status
+    # 2 Get poLine and update payment and receipt status to resolution
     Given path 'orders/order-lines', poLineId
     When method GET
     Then status 200
-
-    * def poLineResponse = $
+    * def poLineResponse = response
     * set poLineResponse.paymentStatus = 'Fully Paid'
     * set poLineResponse.receiptStatus = 'Fully Received'
 
+    # 3. PUT updated PoLine
     Given path 'orders/order-lines', poLineId
     And request poLineResponse
     When method PUT
     Then status 204
 
-  Scenario: Check that order closed
+    # 4. Check that order closed
     Given path 'orders/composite-orders', orderId
-    When method GET
     And retry until response.workflowStatus == 'Closed'
-    Then status 200
-
-    Given path 'orders/composite-orders', orderId
     When method GET
     Then status 200
-    And match $.workflowStatus == 'Closed'
+    And match response.workflowStatus == 'Closed'
 
-  Scenario: Check the encumbrance was released
+    # 5. Check the encumbrance was released
     Given path 'finance/transactions'
     And param query = 'transactionType==Encumbrance and encumbrance.sourcePurchaseOrderId==' + orderId + ' and encumbrance.status==Released'
     When method GET
     Then status 200
-    And match $.totalRecords == 1
+    And match response.totalRecords == 1
+
+    # 5. Delete order and PoLine
+    * table deleteDetails
+      | resourcePath              | resourceId |
+      | 'orders/order-lines'      | poLineId   |
+      | 'orders/composite-orders' | orderId    |
+    * def v = call deleteResource deleteDetails
+
+
+  Scenario: Closed order should not be reopened when it has at least one resolution status
```

Comment:
Can add @Positive or @Negative indicators to the Scenarios that we expect to pass or fail in a controlled-fashion